core/src/main/java/hudson/slaves/CloudRetentionStrategy.java

```diff
@@ -44,36 +45,49 @@
  */
 public class CloudRetentionStrategy extends RetentionStrategy<AbstractCloudComputer> {
     private int idleMinutes;
+    private transient ReentrantLock checkLock;
 
     public CloudRetentionStrategy(int idleMinutes) {
         this.idleMinutes = idleMinutes;
+        readResolve();
```

Comment:
Well, I was basically copying this fix: http://jenkins-ci.org/commit/jclouds-plugin/700644c1bd6d99953042d1a78791b180b1e816d8 after making the same mistake as Kei (https://github.com/jenkinsci/jclouds-plugin/commit/363f3cef682aa2ee4bb34ada46384168989ae2a8). It seems better to have the initialisation code for the ReentrantLock in one place, rather than repeated in different places (unless one was to rely on the RetentionStrategy to always have been deserialized before needing to use the ReentrantLock, but I don't think that would be good design either).